### PR TITLE
feat: tiers of target support

### PR DESCRIPTION
**Pull request by danil-pavlov**

### CODE REVIEW

File: docs/topics/multiplatform/multiplatform-connect-to-apis.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Learn how to [add dependencies on platform-specific libraries](multiplatform-add
 ## Examples
 
 For simplicity, the following examples use intuitive target names, like iOS and Android. However, in your Gradle build files,
-you need to use a specific target name from [the list of supported targets](multiplatform-dsl-reference.md#targets).
+you need to use a specific target name from the [list of supported targets](native-target-support.md).
```
Comment (sbogolepov) [inline docs/topics/multiplatform/multiplatform-connect-to-apis.md:34]: Not a native-specific link.

File: docs/topics/multiplatform/multiplatform-set-up-targets.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [//]: # (title: Set up targets for Kotlin Multiplatform)
 
 You can add targets when [creating a project with the Project Wizard](multiplatform-library.md). If you need to add a target 
-later, you can do this manually using target presets for [supported platforms](multiplatform-dsl-reference.md#targets).
+later, you can do this manually using target presets for [supported platforms](native-target-support.md).
```
Comment (sbogolepov) [inline docs/topics/multiplatform/multiplatform-set-up-targets.md:4]: Not a native-specific link.

File: docs/topics/native/apple-framework.md [hunk 3]
```diff
@@ -145,15 +145,14 @@ binaries {
...
 in the table:
 
 | Target platform/device | Gradle function |
 |------------------------|-----------------|
 | macOS x86_64           | `macosX64()`    | 
-| macOS ARM 64           | `macosArm64()`  | 
-| iOS ARM 32             | `iosArm32()`    | 
+| macOS ARM 64           | `macosArm64()`  |
 | iOS ARM 64             | `iosArm64()`    | 
 | iOS Simulator (x86_64) | `iosX64()`      |
```
Comment (sbogolepov) [inline docs/topics/native/apple-framework.md:157]: And iOS Simulator (arm64) `iosSimulatorArm64`

File: docs/topics/native/native-target-support.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  targets are quite rare.
+
+| Gradle target name   | Target triple                   | Running tests | Description            |
+|----------------------|---------------------------------|---------------|------------------------|
+| `androidNativeArm32` | `arm-unknown-linux-androideabi` |               | Android NDK ARM 32-bit |
+| `androidNativeArm64` | `aarch64-unknown-linux-android` |               | Android NDK ARM 64-bit |
+| `androidNativeX86`   | `i686-unknown-linux-android`    |               | Android NDK x86        |
+| `androidNativeX64`   | `x86_64-unknown-linux-android`  |               | Android NDK x86-64     |
+| `mingwX64`           | `x86_64-pc-windows-gnu`         | ✅             | x86-64 MinGW           |
+|                      |  macOS hosts only              |               |                        |
```
Comment (sbogolepov) [inline docs/topics/native/native-target-support.md:76]:  -> Apple?

File: docs/topics/native/native-command-line-compiler.md [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 The Kotlin/Native compiler is available for macOS, Linux, and Windows. It is available as a command line tool and ships 
 as part of the standard Kotlin distribution and can be downloaded from [GitHub Releases](%kotlinLatestUrl%). It supports
-different targets including iOS (arm32, arm64, simulator x86_64), Windows (mingw32 and x86_64),
+different targets including iOS (arm64, simulator x86_64), Windows (mingw32 and x86_64),
```
Comment (sbogolepov) [inline docs/topics/native/native-command-line-compiler.md:7]: Let's move from explicit enumeration to something like this
> It supports different targets including Linux, macOS, iOS and many others. [See the full list of targets here](native-target-support.md).

**Review by sbogolepov — CHANGES_REQUESTED** [review]

File: docs/topics/multiplatform/multiplatform-dsl-reference.md [hunk 6]
```diff
@@ -82,109 +82,23 @@ The complete list of available targets is the following:
...
-               <li><code>linuxMipsel32</code> — Linux on little-endian MIPS (mipsel) platforms</li>
-               <li><code>linuxX64</code> — Linux on x86_64 platforms</li>
-            </list>
-        </td>
+        <td>Kotlin/Native</td>
+        <td></td>
         <td>
-            <p>Linux MIPS targets (<code>linuxMips32</code> and <code>linuxMipsel32</code>) require a Linux host.</p>
-            <p>You can build other Linux targets on any supported host.</p>
+            <p>Learn about currently supported targets for the macOS, Android NDK, Linux, and Windows hosts in <a href="native-target-support.md">Kotlin/Native target support</a>.</p>
```
Comment (sbogolepov) [inline docs/topics/multiplatform/multiplatform-dsl-reference.md:88]: Android NDK is redundant as it only can be used as a target.

File: docs/topics/native/native-target-support.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    only on macOS x86-64 host.
+
+## Tier 1
+
+* The target is regularly tested on CI to be able to compile and run.
+* We're doing our best to provide a source and [binary compatibility between compiler releases](https://youtrack.jetbrains.com/issue/KT-42293).
+
+| Gradle target name     | Target triple                 | Running tests | Description                                    |
+|------------------------|-------------------------------|---------------|------------------------------------------------|
+| `linuxX64`             | `x86_64-unknown-linux-gnu`    | ✅             | Linux on x86_64 platforms                      |
```
Comment (sbogolepov) [inline docs/topics/native/native-target-support.md:29]: Each (except mingwX64) description cell contains "platforms" word. Maybe we can omit it?

File: docs/topics/native/native-target-support.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+* We can't promise a source and binary compatibility between different compiler releases, though such changes for these
+  targets are quite rare.
+
+| Gradle target name     | Target triple                   | Running tests | Description                                                          |
+|------------------------|---------------------------------|---------------|----------------------------------------------------------------------|
+| `androidNativeArm32`   | `arm-unknown-linux-androideabi` |               | [Android NDK](https://developer.android.com/ndk) on ARM32 platforms  |
+| `androidNativeArm64`   | `aarch64-unknown-linux-android` |               | [Android NDK](https://developer.android.com/ndk) on ARM64 platforms  |
+| `androidNativeX86`     | `i686-unknown-linux-android`    |               | [Android NDK](https://developer.android.com/ndk) on x86 platforms    |
+| `androidNativeX64`     | `x86_64-unknown-linux-android`  |               | [Android NDK](https://developer.android.com/ndk) on x86_64 platforms |
+| `mingwX64`             | `x86_64-pc-windows-gnu`         | ✅             | 64-bit Microsoft Windows                                             |
```
Comment (sbogolepov) [inline docs/topics/native/native-target-support.md:75]: It is not a "real" Windows target (which is MSVC), but a MinGW, which is, roughly speaking, a way to port Linux apps to Windows. It has a different API and ABI, a different set of quirks, so users should know what they use.
So a better description would be something like this:
> 64-bit [MinGW](https://www.mingw-w64.org/) (Windows 7 and newer)

**Review by sbogolepov — CHANGES_REQUESTED** [review]
Overall LGTM. Just need to fix a few small things.

File: docs/topics/native/native-target-support.md (hunk 7 above)
Comment (danil-pavlov) [reply docs/topics/native/native-target-support.md:29]: I'd rather keep it as it makes target details more descriptive

**Review by sbogolepov — APPROVED** [review]

File: docs/topics/multiplatform/multiplatform-dsl-reference.md [hunk 9]
```diff
@@ -82,109 +82,23 @@ The complete list of available targets is the following:
...
     <tr>
-        <td>Windows</td>
+        <td>Android applications and libraries</td>
+        <td><code>android</code></td>
         <td>
-            <list>
-               <li><code>mingwX64</code> — 64-bit Microsoft Windows</li>
-               <li><code>mingwX86</code> — 32-bit Microsoft Windows</li>
-            </list>
+            <p>Manually apply an Android Gradle plugin  – <code>com.android.application</code> or <code>com.android.library</code>.</p>
```
Comment (koshachy) [inline docs/topics/multiplatform/multiplatform-dsl-reference.md:95]:
```suggestion
            <p>Manually apply an Android Gradle plugin: <code>com.android.application</code> or <code>com.android.library</code>.</p>
```

File: docs/topics/native/native-target-support.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+The Kotlin/Native compiler supports a great number of different targets, though it is hard to provide the same level of
+support for all of them. This document describes which targets Kotlin/Native supports and breaks them into several
+tiers depending on how well the compiler supports them.
+
+> We can adjust the number of tiers, the list of supported targets, and their features as we go.
+> 
+{type="note"}
+
+Mind the following terms used in tier tables:
```
Comment (koshachy) [inline docs/topics/native/native-target-support.md:11]: Maybe also make it a note (gray one)? 🤔

**Review by koshachy — APPROVED** [review]
A bit out of the whole Kotlin docs, but it's OK.

**Pull request merged**